src/main/java/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -134,7 +136,7 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
         val types = listOf(superclass).filter { it != ANY }.map {
           CodeBlock.of("%T(%L)", it, superclassConstructorParameters.joinToCode())
         }
-        val superTypes = types + superinterfaces.map { CodeBlock.of("%T", it) }
+        val superTypes = types + superinterfaces.map { CodeBlock.of("%T", it) } + superinterfaceDelegates.map { CodeBlock.of("${it.type} by ${it.name}") }
```

Comment:
Good call. I was pretty sure that there'd be something more to this because it's dealing with types. Recurse up the type hierarchy is the only way I think.  There should be already something like this in kotlinpoet for the rest of the type names though? I think I saw it when writing this